### PR TITLE
refactor(behavior_velocity_planner_common): simplify header files

### PR DESCRIPTION
**Pull request by soblin**
## Description

https://github.com/autowarefoundation/autoware.universe/issues/4821

## Tests performed

Psim works

memory usage during the build is improved

image

## Interface changes

none

## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kyoichi-sugahara — APPROVED** [review]

File: planning/behavior_velocity_crosswalk_module/include/behavior_velocity_crosswalk_module/util.hpp [hunk 1]
```diff
@@ -35,6 +35,9 @@
 #include <Eigen/Geometry>
 #include <behavior_velocity_planner_common/planner_data.hpp>
 
+// TODO(Mamoru Sobue): keep this temporarily
```
Comment (TakaHoribe) [inline planning/behavior_velocity_crosswalk_module/include/behavior_velocity_crosswalk_module/util.hpp:38]: Should we still keep this?

**Review by maxime-clem — APPROVED** [review]
Tested the branch and there is no issue after the refactoring.

File: planning/behavior_velocity_planner_common/src/scene_module_interface.cpp [hunk 2]
```diff
@@ -0,0 +1,264 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/scene_module_interface.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: planning/behavior_velocity_planner_common/src/utilization/arc_lane_util.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/utilization/arc_lane_util.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: planning/behavior_velocity_planner_common/src/utilization/boost_geometry_helper.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/utilization/boost_geometry_helper.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: planning/behavior_velocity_planner_common/src/utilization/trajectory_utils.cpp [hunk 5]
```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/utilization/trajectory_utils.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

File: planning/behavior_velocity_planner_common/src/velocity_factor_interface.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 Tier IV, Inc.
```
Comment (isamu-takagi) [inline planning/behavior_velocity_planner_common/src/velocity_factor_interface.cpp:1]:
```suggestion
// Copyright 2023 TIER IV, Inc.
```

**Review by TomohitoAndo — APPROVED** [review]

File: planning/behavior_velocity_planner_common/src/scene_module_interface.cpp (hunk 2 above)
Comment (soblin) [reply planning/behavior_velocity_planner_common/src/scene_module_interface.cpp:1]: fixed

File: common/tier4_autoware_utils/src/geometry/geometry.cpp [hunk 7]
```diff
@@ -17,6 +17,13 @@
 #include <Eigen/Geometry>
 
 #include <tf2/convert.h>
+
+#ifdef ROS_DISTRO_GALACTIC
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#else
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#endif
+
```
[the remark this answers is not in the document]
Comment (soblin) [reply common/tier4_autoware_utils/src/geometry/geometry.cpp:26]: fixed

File: planning/behavior_velocity_crosswalk_module/include/behavior_velocity_crosswalk_module/util.hpp (hunk 1 above)
Comment (soblin) [reply planning/behavior_velocity_crosswalk_module/include/behavior_velocity_crosswalk_module/util.hpp]: This was by mistake, not fixed

**Review by isamu-takagi — APPROVED** [review]

File: common/tier4_autoware_utils/src/geometry/geometry.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (TakaHoribe) [reply common/tier4_autoware_utils/src/geometry/geometry.cpp:26]: @soblin seems not fixed yet?

**Review by TakaHoribe — APPROVED** [review]

File: common/tier4_autoware_utils/src/geometry/geometry.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (soblin) [reply common/tier4_autoware_utils/src/geometry/geometry.cpp]: I fixed this with the latest commit

**Review by shmpwk — APPROVED** [review]

**Pull request merged**